Refactor namespace matching logic across config subsystem

pjcodes404 opened the pull request:
## Summary

This PR refactors the namespace matching logic across the `clu.config` subsystem.  
It introduces a centralized helper function `namespace_matches()` to replace brittle
`str.startswith()`-based matching, ensuring correct and consistent namespace behavior.

The new logic handles:
- Exact matches (e.g., "foo" == "foo")
- Hierarchical matches (e.g., "foo:bar" starts with "foo")
- Rejects false positives (e.g., "foobar" ≠ "foo", "foo.barista" ≠ "foo.bar")

This eliminates incorrect partial matches and unifies namespace handling across
multiple modules.

---

## Changes

### 1. New Helper Function
Added `namespace_matches(namespace, prefix)` in `clu/config/ns.py`, restoring the file
structure and fixing indentation issues.

### 2. Refactored All Namespace Logic
Replaced `str.startswith()` and related prefix checks with `namespace_matches()` in:

- `clu/config/abc.py`  
  - `FrozenKeyMap.submap` now uses the helper
- `clu/config/keymap.py`  
  - `Flat.keys()` and `Flat.values()` updated
- `clu/config/legacy/base.py`  
  - Updated namespace filtering logic
- `clu/config/keymapview.py`  
  - All `NamespaceWalker*View` classes now use the helper

### 3. Bug Fixes
- Fixed `IndentationError` and file corruption in `ns.py`
- Fixed `SyntaxError` in `abc.py` caused by smart quotes
- Removed duplicate and dead code in `keymapview.py`

---

## Testing & Verification

Added a comprehensive test suite in `tests/test_namespace_matching.py`
covering:

- `namespace_matches()` exact, prefix, and negative cases  
- `startswith_ns` wrapper behavior  
- `FrozenKeyMap.submap` integration  
- `Flat.keys()` / `Flat.values()` namespace filtering  
- NamespaceWalker views (`KeysView`, `ItemsView`, `ValuesView`)

All tests passed successfully:


## Why This Matters
The previous namespace logic allowed partial prefix collisions, which could
produce incorrect lookups and unexpected behavior in real-world configurations.
By centralizing the logic into `namespace_matches()`, all namespace-aware
structures now behave consistently and predictably.

fish2000 commented:
This stuff is all great, but can you resubmit it without all the changes to existing docstrings? Things like:

![image](https://github.com/user-attachments/assets/b759721e-6adb-40c1-96b3-772d9c769a10)

… aren't related to the primary change in the PR.

If you remove those edits I'll see about integrating this, like I said it looks terrific @pjcodes404 – thanks

fish2000 commented on tests/test_namespace_matching.py:
```diff
@@ -0,0 +1,155 @@
+        self.assertIn(1, values)
+        self.assertIn(2, values)
+        self.assertNotIn(3, values)
+
```
I super appreciate these tests, and they seem well-written – please consider updating these to our house-standard `py.test` syntax and/or using CLU's inline test harness!

fish2000 commented on clu/config/abc.py:
```diff
@@ -183,12 +183,12 @@ def submap(self, *namespaces, unprefixed=False):
     def keys(self, *namespaces, unprefixed=False):
         """ Return a namespaced view over either all keys in the mapping,
-            or over only those keys in the mapping matching the specified
+            or over only those keys in the mapping which match the specified
```
I particularly appreciate this typo fix

fish2000 commented:
This is fantastic, thank you! I'll run everything on my end, but your contributions are very much appreciated and will be going forward, yes! Will get these in soon assuredly.

pjcodes404 commented:
> This is fantastic, thank you! I'll run everything on my end, but your contributions are very much appreciated and will be going forward, yes! Will get these in soon assuredly.

Thank you so much! Really appreciate your guidance and patience throughout the
review. I’m happy everything looks good on your end.

If there’s anything further you'd like me to help with—cleanup, follow-up
refactors, or new issues—I'd be happy to contribute more. 🙌

pjcodes404 commented:
> > > This is fantastic, thank you! I'll run everything on my end, but your contributions are very much appreciated and will be going forward, yes! Will get these in soon assuredly.
> > 
> > 
> > Thank you so much! Really appreciate your guidance and patience throughout the review. I’m happy everything looks good on your end.
> > If there’s anything further you'd like me to help with—cleanup, follow-up refactors, or new issues—I'd be happy to contribute more. 🙌
> 
> Absolutely, thanks for saying so! It’s all up. I very slightly rearranged the file containing the tests you wrote – I added your name to the programmer notes within and renamed it.
> 
> It’d be truly fantastic if you were to deign to contribute more, of course! If you encounter problems with CLU’s use of high-codepoint unicode characters in things like docstrings or indenting, do not hesitate to report these as bugs.
> 
> If you have any experience setting up `travis`/`codecov`/other CI GitHub stuff, a hand with that would go a long way. I’ll post an issue for this myself. EDIT: I did so, see Issue #23
> 
> And if anything comes to mind do let me know what your thoughts are. Thanks again my friend!

Thanks again! I'm a bit tied up for the next day or two, but after that I'll
definitely take a look at Issue #23 and the other areas you suggested. Really
appreciate the warm welcome — excited to continue contributing once I'm free.
